File: src/commands/exec.ts
import chalk from 'chalk';
import executeScript from '../lib/executeScript.js';
import isProjectInitialized from '../lib/isProjectInitialized.js';

async function execCommand(args: string[]): Promise<void> {
    if (!isProjectInitialized()) {
        console.log(chalk.red('This command require a nautus project. Initialize it using ' + chalk.cyan('nautus create') + '!'));
        return;
    }
    
    if (typeof args[0] !== 'string' || args[0].length < 1) {
        console.log(chalk.red('Error: Please provide a script name'));
        console.log(chalk.cyan('Example: nautus exec Run'));
        process.exit(1);
    }

    await executeScript(args[0]);
}

export default [execCommand, 'exec <script>', 'Run specific script'] as const;

File: src/commands/exec.rs
use anyhow::{Context, Result};
use colored::Colorize;
use crate::utils;
use crate::js_executor;

pub async fn execute(script: &str) -> Result<()> {
    if !utils::is_project_initialized() {
        println!("{}", "This command requires a nautus project. Initialize it using nautus create!".red());
        return Ok(());
    }

    println!("{}", format!("Executing script: @{}.js", script).cyan());
    js_executor::execute_nautus_script(script, false)
        .await
        .context("Failed to execute script")?;

    Ok(())
}
